tests/ui/unnecessary_reserve.rs

```diff
@@ -0,0 +1,95 @@
+#![warn(clippy::unnecessary_reserve)]
+#![feature(custom_inner_attributes)]
+
+use std::collections::HashMap;
+use std::collections::VecDeque;
+
+fn main() {
+    vec_reserve();
+    vec_deque_reserve();
+    hash_map_reserve();
+    msrv_1_62();
+}
+
+fn vec_reserve() {
+    let mut vec: Vec<usize> = vec![];
+    let array: &[usize] = &[1, 2];
+
+    // do lint
+    vec.reserve(1);
+    vec.extend([1]);
```

Comment:
I'm unsure if this should lint. I think this should only lint for

```rust
vec.reserve(expr.len());
vec.extend(expr);
```

where `expr` is just any arbitrary expression where `len` is called on in the `reserve` call and then used in the `extend` call. To check if the expression is the same, you can use `SpanlessEq`.

Or am I missing something?